### DISCUSSION TITLE
Shards failed Type=circuit_breaking_exception

### DISCUSSION THREAD

**Discussion by juddcbrown** [Security-Onion-Solutions/securityonion#5260] [Unsupported Versions]
I am getting failed shards on Alert with the the following reasons. I have been unable to find a root case or a fix. Please?

{
  "took": 10282,
  "timed_out": false,
  "_shards": {
    "total": 433,
    "successful": 428,
    "skipped": 413,
    "failed": 5,
    "failures": [
      {
        "shard": 2,
        "index": "xxx_searchnode:so-zeek-2021.08.22",
        "node": "gw241BuYSXGiE-zzPTC2LA",
        "reason": {
          "type": "circuit_breaking_exception",
          "reason": "[parent] Data too large, data for [preallocate[aggregations]] would be [5443091840/5gb], which is larger than the limit of [5403102412/5gb], real usage: [5443085696/5gb], new bytes reserved: [6144/6kb], usages [request=80312/78.4kb, fielddata=1970975/1.8mb, in_flight_requests=1369606/1.3mb, model_inference=0/0b, accounting=661545140/630.8mb]",
          "bytes_wanted": 5443091840,
          "bytes_limit": 5403102412,
          "durability": "PERMANENT"
        }
      },
      {
        "shard": 2,
        "index": "xxx-search_searchnode:so-zeek-2021.08.23",
        "node": "gw241BuYSXGiE-zzPTC2LA",
        "reason": {
          "type": "circuit_breaking_exception",
          "reason": "[parent] Data too large, data for [preallocate[aggregations]] would be [5438897536/5gb], which is larger than the limit of [5403102412/5gb], real usage: [5438891392/5gb], new bytes reserved: [6144/6kb], usages [request=63872/62.3kb, fielddata=1970975/1.8mb, in_flight_requests=1378526/1.3mb, model_inference=0/0b, accounting=661545140/630.8mb]",
          "bytes_wanted": 5438897536,
          "bytes_limit": 5403102412,
          "durability": "PERMANENT"
        }
      },
      {
        "shard": 3,
        "index": "xxx-search_searchnode:so-zeek-2021.08.23",
        "node": "gw241BuYSXGiE-zzPTC2LA",
        "reason": {
          "type": "circuit_breaking_exception",
          "reason": "[parent] Data too large, data for [preallocate[aggregations]] would be [5459869056/5gb], which is larger than the limit of [5403102412/5gb], real usage: [5459862912/5gb], new bytes reserved: [6144/6kb], usages [request=63872/62.3kb, fielddata=1970975/1.8mb, in_flight_requests=1460502/1.3mb, model_inference=0/0b, accounting=661545140/630.8mb]",
          "bytes_wanted": 5459869056,
          "bytes_limit": 5403102412,
          "durability": "PERMANENT"
        }
      },
      {
        "shard": 4,
        "index": "xxx-search_searchnode:so-zeek-2021.08.22",
        "node": "gw241BuYSXGiE-zzPTC2LA",
        "reason": {
          "type": "circuit_breaking_exception",
          "reason": "[parent] Data too large, data for [indices:data/read/search[phase/query]] would be [5447281244/5gb], which is larger than the limit of [5403102412/5gb], real usage: [5447280000/5gb], new bytes reserved: [1244/1.2kb], usages [request=57728/56.3kb, fielddata=1970975/1.8mb, in_flight_requests=1528554/1.4mb, model_inference=0/0b, accounting=661545140/630.8mb]",
          "bytes_wanted": 5447281244,
          "bytes_limit": 5403102412,
          "durability": "PERMANENT"
        }
      },
      {
        "shard": 4,
        "index": "xxx-search_searchnode:so-zeek-2021.08.23",
        "node": "gw241BuYSXGiE-zzPTC2LA",
        "reason": {
          "type": "circuit_breaking_exception",
          "reason": "[parent] Data too large, data for [indices:data/read/search[phase/query]] would be [5459864156/5gb], which is larger than the limit of [5403102412/5gb], real usage: [5459862912/5gb], new bytes reserved: [1244/1.2kb], usages [request=57728/56.3kb, fielddata=1970975/1.8mb, in_flight_requests=1633338/1.5mb, model_inference=0/0b, accounting=661545140/630.8mb]",
          "bytes_wanted": 5459864156,
          "bytes_limit": 5403102412,
          "durability": "PERMANENT"
        }
      }
    ]
  },
  "_clusters": {
    "total": 2,
    "successful": 2,
    "skipped": 0
  },
  "hits": {
    "total": 104106,
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "2": {
      "doc_count_error_upper_bound": 8,
      "sum_other_doc_count": 2181,
      "buckets": [
        {
          "key": "x.x.x.x",
          "doc_count": 30742
        },
        {
          "key": "x.x.x.x",
          "doc_count": 25128
        },
        {
          "key": "x.x.x.x",
          "doc_count": 8159
        },
        {
          "key": "x.x.x.x",
          "doc_count": 7080
        },
        {
          "key": "x.x.x.x",
          "doc_count": 7080
        },
        {
          "key": "x.x.x.x",
          "doc_count": 3836
        },
        {
          "key": "x.x.x.x",
          "doc_count": 2669
        },
        {
          "key": "x.x.x.x",
          "doc_count": 1416
        },
        {
          "key": "x.x.x.x",
          "doc_count": 1328
        },
        {
          "key": "x.x.x.x",
          "doc_count": 1064
        },
        {
          "key": "x.x.x.x",
          "doc_count": 914
        },
        {
          "key": "x.x.x.x",
          "doc_count": 813
        },
        {
          "key": "x.x.x.x",
          "doc_count": 775
        },
        {
          "key": "x.x.x.x",
          "doc_count": 763
        },
        {
          "key": "x.x.x.x",
          "doc_count": 761
        },
       .......
        }
      ]
    }
  }
}

**Comment by csanate** [comment]
I'm struggling with the same issue. 

I have a manager, two search nodes and a forward sensor. In my case, the search nodes have 16Gb of RAM each, but they are only using about 8. I'm trying to identify how to update elastic to use the additional ram. I'll post here if I find a solution. 

Docs on ES heap size: https://docs.securityonion.net/en/2.3/elasticsearch.html#heap-size

>If total available memory is 8GB or greater, Setup configures the heap size to be 33% of available memory, but no greater than 25GB. >You may need to adjust the value for heap size depending on your system’s performance. This can be modified in /opt/so/saltstack/local/pillar/minions/$minion.sls.

The figure of 33% seems low for a dedicated node, but I'm not really a expert in Elastic Search... Based on this, I increased the heap size to use more memory. It'll be a few days before I know if the search nodes have stopped working again...

**Reply by juddcbrown** [reply]
Csanate, thanks in advance.

Of equal concern to me is receiving no response whatever for 21 days. Is it due to improper post formatting? My last 3 posts have received no response. 

BTW I think this issue may relate to all machines being successfully at high state. I’m still being asked to reboot devices days after soup, and salt \* state.high state is failing on 1 or 2 devices. I see the shard issue most often when that is the case.

Thanks, Judd

**Reply by csanate** [reply]
Yeah, I'm getting the reboot notices as well. 

Not sure if it's related or if the reboot is helping with the ES issue as a side effect. 

I think support on here is minimal at best... but hey compared to the cost of most IDS solutions... I guess we get what we pay for? LOL I'll let you know how the fix goes... so far so good...

**Reply by TOoSmOotH** [reply]
@juddcbrown The reason you probably didn't get a response is because this has been answered before. If you searched for circuit_breaking_exception in the discussions search, you would have seen the result for #4172 which gives a detailed explanation.  Please read #1720 as well. Specifically: 

**Community Support**
Community Support is considered best effort and there are no guarantees and no SLAs. If you need private or priority support, please consider purchasing support from [Security Onion Solutions](https://securityonionsolutions.com/support).

**Comment by juddcbrown** [comment]
Mike, thank you for the explanation. I’ll search more thoroughly in the
future.

On Mon, Sep 13, 2021 at 15:15 Mike Reeves ***@***.***> wrote:

> @juddcbrown <https://github.com/juddcbrown> The reason you probably
> didn't get a response is because this has been answered before. If you
> searched for circuit_breaking_exception in the discussions search, you
> would have seen the result for #4172
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/4172>
> which gives a detailed explanation. Please read #1720
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/1720>
> as well. Specifically:
>
> *Community Support*
> Community Support is considered best effort and there are no guarantees
> and no SLAs. If you need private or priority support, please consider
> purchasing support from Security Onion Solutions
> <https://securityonionsolutions.com/support>.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/5260#discussioncomment-1320461>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH4XT7TKWAZX4DRX37Y5A6DUBZLV7ANCNFSM5CUUHU7A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Judd Brown
